### PR TITLE
Fixes OOM (Java Heap Space) when Unzipping by Writing to File if necessary + Refactoring

### PR DESCRIPTION
**Pull request by fhaScireum**
(non breaking)

- Fixes: OX-6679

### CODE REVIEW

File: src/main/java/sirius/biz/util/ExtractedFileBuffer.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * Wraps a {@link DeferredFileOutputStream stream} that contains an extracted file either retained in memory or in
+ * a temporary file.
+ */
+public class ExtractedFileBuffer {
```
Comment (andyHa) [inline src/main/java/sirius/biz/util/ExtractedFileBuffer.java:25]: implement Closable?

**Review by idlira — APPROVED** [review]

File: src/main/java/sirius/biz/util/ExtractedFileBuffer.java (hunk 1 above)
Comment (fhaScireum) [reply src/main/java/sirius/biz/util/ExtractedFileBuffer.java:25]: I thought about that, but we only close the output stream but no input streams. I'm not sure if that would confuse people, but i can change that if you'd like.

**Review by andyHa — APPROVED** [review]
Looks generally good, maybe implement Closable to earn extra checks by Java/inspections. Or make this class package private as it shouldn't be used outside....

File: src/main/java/sirius/biz/util/ExtractedFileBuffer.java (hunk 1 above)
Comment (andyHa) [reply src/main/java/sirius/biz/util/ExtractedFileBuffer.java:25]: make this class package private then (?)

Comment (fhaScireum) [reply src/main/java/sirius/biz/util/ExtractedFileBuffer.java]: Done

**Review by sabieber — APPROVED** [review]

**Review by mkeckmkeck — APPROVED** [review]

**Review by cschierle — APPROVED** [review]

**Pull request merged**